src/CloneExtensions/ExpressionFactories/ComplexTypeExpressionFactory.cs

```diff
@@ -198,6 +190,29 @@ private IEnumerable<FieldInfo> GetBackingFields(Type type)
             }
         }
 
+        private static IEnumerable<PropertyInfo> GetProperties(Type type)
+        {
+            TypeInfo typeInfo = type.GetTypeInfo();
+
+            while (typeInfo != null && typeInfo.UnderlyingSystemType != _objectType)

```

Comment:
It just occurred to me. Will that work for value types? We probably want to stop when we hit `ValueType` class. Both here and in `GetBackingFields`.

Reply:
"Will that work for value types?" 

I am not sure what you mean by "that".  I do not think value types, structs, can subclass other classes ... implement interfaces yes, but I do not think they can subclass.

-Jeff

Reply:
They can't derive from other structs, true. But they all derive from `ValueType`. I'm wondering if in our walk down the inheritance chain we should stop when we meet `ValueType`. But, `ValueType` does not define any properties/fields, so actually that doesn't matter after all.